lib/index.js

```diff
@@ -12,54 +12,55 @@ const promiseSpawn = (cmd, args, opts = {}, extra = {}) => {
     return spawnWithShell(cmd, args, opts, extra)
   }
 
-  let proc
+  let resolve, reject
+  const promise = new Promise((_resolve, _reject) => {
+    resolve = _resolve
+    reject = _reject
+  })
 
-  const p = new Promise((res, rej) => {
-    proc = spawn(cmd, args, opts)
+  // Create error here so we have a more useful stack trace when rejecting
+  const closeError = new Error('command failed')
```

Comment:
This is handy but the stack trace always points back just to this line, not to lines 43, 47, 52, or 57 which would be much more useful to know.  Did we reject cause of a proc error?  stdin error? Code or signal?

Reply:
It's "4x" as much code but we should really be generating the error object in each of those places.  It's worth it.

Reply:
I'm not sure I follow. This `closeError` is only used when rejecting via `proc.on('close')` on line 57. Since line 57 is an async context it gets cut off from the parents stack trace. In this case I do think it's preferable to get the stack trace from line 22 rather than line 57.

I don't actually know what the stack traces look like for the other error events. I was only concerned with processes that close via error code or signal since I believe that's the vast majority of errors.

Reply:
Oh yes I see it now 

```
proc.stdout.on('error', rejectWithOpts)
```

proc will bring its own error here.  I think generating an error in line 57 may also get us back to the problem you're solving in the first place.  Nevermind.